### PR TITLE
Use a prototype TextBlock to resolve theme resources on Windows

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change ###

On the Windows platforms, the device styles were being retrieved and copied into the Forms resource dictionary by walking up the style tree and retrieving the Setter values. However, this fails when the walk up the style tree encounters a Setter whose value is a binding expression (e.g., to ThemeResource). Because of this, most of the built-in device styles do not render at all on the Windows platforms.

Rather than walk the tree, this change creates a TextBlock, applies the style to it, and retrieves the resolved values for use in the Forms resource dictionary. This allows it to retrieve values which come from ThemeResource bindings. The rendered device styles now match more closely with the values in WinRT apps created in Blend:

[Windows Phone 8.1 Styles before](https://www.dropbox.com/s/4vvwoi32uk6jemp/wp81_before.png?raw=1) 
[Windows Phone 8.1 Styles from Blend](https://www.dropbox.com/s/posfbze82rnqpb5/wp81_blend.png?raw=1)
[Windows Phone 8.1 Styles after](https://www.dropbox.com/s/kbaf7r3y8rrz2zr/wp81_after.png?raw=1)

[Windows 8.1 Styles Before](https://www.dropbox.com/s/nz7ehlve3kgpj45/w81_before.PNG?raw=1)
[Windows 8.1 Styles from Blend](https://www.dropbox.com/s/jif9imaqijbp6fg/w81_blend.PNG?raw=1)
[Windows 8.1 Styles After](https://www.dropbox.com/s/gix3zqy7vksetdi/w81_after.PNG?raw=1)

This also applies to UWP, but I got tired of taking screenshots.

### Bugs Fixed ###

- [43783 – [WP8.1] Most Device Styles do not render correctly in Windows Phone 8.1 (RT) applications](https://bugzilla.xamarin.com/show_bug.cgi?id=43783)

### API Changes ###

- None

### Behavioral Changes ###

- Anyone who previously relied on the incorrect device styles would need to update their code (though the updates would mostly consist of simply removing the device style). 

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Platform.WinRT.Phone/WindowsPhoneResourcesProvider.cs [hunk 1]
```diff
@@ -12,89 +10,50 @@ internal sealed class WindowsPhoneResourcesProvider
...
-			{
-				BasedOn = value.BasedOn
-			};
-			foreach (var valSetter in value.Setters)
-			{
-				style.Setters.Add(valSetter);
-			}
-
-			var ancestorStyles = new List<Windows.UI.Xaml.Style> ();
+		readonly TextBlock _prototype = new TextBlock();
```
Comment (samhouts) [inline Xamarin.Forms.Platform.WinRT.Phone/WindowsPhoneResourcesProvider.cs]: Should we put this nearer the top? Seems weird here in the middle.

File: Xamarin.Forms.Platform.WinRT.Tablet/WindowsResourcesProvider.cs [hunk 2]
```diff
@@ -15,60 +15,47 @@ internal sealed class WindowsResourcesProvider : ISystemResourcesProvider
...
+			formsStyle.Setters.Add(Label.FontSizeProperty, _prototype.FontSize);
+			formsStyle.Setters.Add(Label.FontFamilyProperty, _prototype.FontFamily.Source);
+			formsStyle.Setters.Add(Label.FontAttributesProperty, ToAttributes(_prototype.FontWeight));
+			formsStyle.Setters.Add(Label.LineBreakModeProperty, ToLineBreakMode(_prototype.TextWrapping));
 
 			return formsStyle;
 		}
 
-		static FontAttributes ToAttributes(ushort uweight)
+		readonly TextBlock _prototype = new TextBlock();
```
Comment (samhouts) [inline Xamarin.Forms.Platform.WinRT.Tablet/WindowsResourcesProvider.cs]: Same question...should this go near the top of the file?

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Bugzilla43783.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
+﻿using Xamarin.Forms.CustomAttributes;
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.Forms.Controls.Issues
+{
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Bugzilla, 43783, "[WP8.1] Most Device Styles do not render correctly in Windows Phone 8.1 (RT) applications", PlatformAffected.WinPhone)]
```
Comment (samhouts) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Bugzilla43783.cs]: Looks like these code changes also affect non-phones, right?

Comment (hartez) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Bugzilla43783.cs]: Yes. The issue exists on all WinRT/UWP platforms.

File: Xamarin.Forms.Platform.WinRT.Phone/WindowsPhoneResourcesProvider.cs (hunk 1 above)
Comment (hartez) [inline Xamarin.Forms.Platform.WinRT.Phone/WindowsPhoneResourcesProvider.cs]: Yes. Done.

**Pull request merged**